test/framework/src/main/java/org/elasticsearch/transport/nio/InboundChannelBuffer.java

```diff
@@ -30,39 +33,48 @@
  * the pages internally. If more space is needed at the end of the buffer {@link #ensureCapacity(long)} can
  * be called and the buffer will expand using the supplier provided.
  */
-public final class InboundChannelBuffer {
+public final class InboundChannelBuffer implements Releasable {
 
     private static final int PAGE_SIZE = 1 << 14;
     private static final int PAGE_MASK = PAGE_SIZE - 1;
     private static final int PAGE_SHIFT = Integer.numberOfTrailingZeros(PAGE_SIZE);
     private static final ByteBuffer[] EMPTY_BYTE_BUFFER_ARRAY = new ByteBuffer[0];
 
 
-    private final ArrayDeque<ByteBuffer> pages;
-    private final Supplier<ByteBuffer> pageSupplier;
+    private final ArrayDeque<Page> pages;
+    private final Supplier<Page> pageSupplier;
 
     private long capacity = 0;
     private long internalIndex = 0;
     // The offset is an int as it is the offset of where the bytes begin in the first buffer
     private int offset = 0;
 
     public InboundChannelBuffer() {
-        this(() -> ByteBuffer.wrap(new byte[PAGE_SIZE]));
```

Comment:
can we remove this default ctor and pass the non-recycling closure in the test directly. I think it's just one place.